lib/ash/actions/managed_relationships.ex

```diff
@@ -1727,30 +1727,30 @@ defmodule Ash.Actions.ManagedRelationships do
 
         if field == relationship.destination_attribute do
           if is_struct(input) do
-            do_matches?(current_value, input, field, attr.type)
+            do_matches?(current_value, input, field, attr.type, attr.constraints)
```

Comment:
I was not sure if I should add the constraints as a separate param or pass the whole `attr` instead of `type` and extract type and constraints later. Let me know if this is preferred.